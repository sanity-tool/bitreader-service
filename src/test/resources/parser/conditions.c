static int I, *I_P;

void foo();
void bar();

static void testImplicitIf() {
    if (I) {
        foo();
    }
}

static void testImplicitIfElse() {
    if (I) {
        foo();
    } else {
        bar();
    }
}

static void testExplicitIf() {
    if (I != 0) {
        foo();
    }
}
