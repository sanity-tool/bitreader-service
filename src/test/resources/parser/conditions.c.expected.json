{
  "functions" : [ {
    "ref" : {
      "kind" : "Global",
      "index" : 0
    },
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMICmp",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 0
        }, {
          "kind" : "Global",
          "index" : 1
        } ],
        "sourceRef" : 0,
        "predicate" : "LLVMIntNE"
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 1
        }, {
          "kind" : "Block",
          "index" : -1
        }, {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 0
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Global",
          "index" : 3
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 2
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMRet",
        "typeId" : 2,
        "sourceRef" : 3
      } ]
    } ],
    "entryBlockIndex" : 0
  }, {
    "ref" : {
      "kind" : "Global",
      "index" : 4
    },
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 4
      }, {
        "kind" : "LLVMICmp",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 0
        }, {
          "kind" : "Global",
          "index" : 1
        } ],
        "sourceRef" : 4,
        "predicate" : "LLVMIntNE"
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 1
        }, {
          "kind" : "Block",
          "index" : -1
        }, {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 4
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Global",
          "index" : 3
        } ],
        "sourceRef" : 5
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 6
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Global",
          "index" : 5
        } ],
        "sourceRef" : 7
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : -1
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMRet",
        "typeId" : 2,
        "sourceRef" : 8
      } ]
    } ],
    "entryBlockIndex" : 0
  }, {
    "ref" : {
      "kind" : "Global",
      "index" : 6
    },
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 9
      }, {
        "kind" : "LLVMICmp",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 0
        }, {
          "kind" : "Global",
          "index" : 1
        } ],
        "sourceRef" : 9,
        "predicate" : "LLVMIntNE"
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 1
        }, {
          "kind" : "Block",
          "index" : -1
        }, {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 9
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Global",
          "index" : 3
        } ],
        "sourceRef" : 10
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 11
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMRet",
        "typeId" : 2,
        "sourceRef" : 12
      } ]
    } ],
    "entryBlockIndex" : 0
  } ],
  "types" : [ {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 1 ]
  }, {
    "kind" : "LLVMFunctionTypeKind",
    "types" : [ 2 ]
  }, {
    "kind" : "LLVMVoidTypeKind"
  }, {
    "kind" : "LLVMIntegerTypeKind",
    "len" : 32
  }, {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 3 ]
  }, {
    "kind" : "LLVMIntegerTypeKind",
    "len" : 1
  }, {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 7 ]
  }, {
    "kind" : "LLVMFunctionTypeKind",
    "types" : [ 2 ]
  } ],
  "globals" : [ {
    "kind" : "LLVMFunctionValueKind",
    "name" : "testImplicitIf",
    "typeId" : 0
  }, {
    "kind" : "LLVMConstantIntValueKind",
    "typeId" : 3,
    "intValue" : 0
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "I",
    "typeId" : 4,
    "operands" : [ {
      "kind" : "Global",
      "index" : 1
    } ]
  }, {
    "kind" : "LLVMFunctionValueKind",
    "name" : "foo",
    "typeId" : 6
  }, {
    "kind" : "LLVMFunctionValueKind",
    "name" : "testImplicitIfElse",
    "typeId" : 0
  }, {
    "kind" : "LLVMFunctionValueKind",
    "name" : "bar",
    "typeId" : 6
  }, {
    "kind" : "LLVMFunctionValueKind",
    "name" : "testExplicitIf",
    "typeId" : 0
  } ],
  "sourceRefs" : [ {
    "file" : "parser/conditions.c",
    "line" : 7
  }, {
    "file" : "parser/conditions.c",
    "line" : 8
  }, {
    "file" : "parser/conditions.c",
    "line" : 9
  }, {
    "file" : "parser/conditions.c",
    "line" : 10
  }, {
    "file" : "parser/conditions.c",
    "line" : 13
  }, {
    "file" : "parser/conditions.c",
    "line" : 14
  }, {
    "file" : "parser/conditions.c",
    "line" : 15
  }, {
    "file" : "parser/conditions.c",
    "line" : 16
  }, {
    "file" : "parser/conditions.c",
    "line" : 18
  }, {
    "file" : "parser/conditions.c",
    "line" : 21
  }, {
    "file" : "parser/conditions.c",
    "line" : 22
  }, {
    "file" : "parser/conditions.c",
    "line" : 23
  }, {
    "file" : "parser/conditions.c",
    "line" : 24
  } ]
}